### PR TITLE
Fixes #678

### LINKED ISSUE #678: ASCompletion explicit scope - broken
ASCompletion: Generate explicit scope is set to True.
Using "Promote to class member" on button, result is:
button = new Button();
instead of this.button

Using "Generate Event handler" code generator, result is
      private function onAddedToStage(e:Event):void {
         removeEventListener(Event.ADDED_TO_STAGE, onAddedToStage);
Expected this.removeEventListener(Event.ADDED_TO_STAGE, onAddedToStage);

http://www.flashdevelop.org/community/viewtopic.php?f=6&t=12180

### PR DESCRIPTION
**Pull request by SlavaRa**

### CODE REVIEW

**Comment by SlavaRa** [comment]
@elsassph @Neverbirth guys, please review this PR

File: External/Plugins/ASCompletion/Completion/ASGenerator.cs [hunk 1]
```diff
@@ -769,11 +770,11 @@ private static void ShowFieldFromParameter(FoundDeclaration found, List<IComplet
                 if (GetDefaultVisibility(found.inClass) == Visibility.Protected)
                     label = TextHelper.GetString("ASCompletion.Label.GenerateProtectedFieldFromParameter");
                 else label = TextHelper.GetString("ASCompletion.Label.GeneratePrivateFieldFromParameter");
-                options.Add(new GeneratorItem(label, GeneratorJobType.FieldFromPatameter, found.member, found.inClass, parameters));
+                options.Add(new GeneratorItem(label, GeneratorJobType.FieldFromParameter, found.member, found.inClass, parameters));
```
Comment (elsassph) [inline External/Plugins/ASCompletion/Completion/ASGenerator.cs]: LOL "Patameter"

File: External/Plugins/ASCompletion/Completion/ASGenerator.cs [hunk 2]
```diff
@@ -2565,14 +2567,21 @@ private static List<FunctionParameter> ParseFunctionParameters(ScintillaControl
...
-            bool detach, ClassModel inClass)
+        private static void GenerateFunctionJob(GeneratorJobType job, ScintillaControl sci, MemberModel member, bool detach, ClassModel inClass)
         {
-            int position = 0;
-            MemberModel latest = null;
+            var position = 0;
+            if (ASContext.CommonSettings.GenerateScope)
+            {
+                position = sci.CurrentPos;
+                var start = sci.WordStartPosition(position, true);
```
Comment (elsassph) [inline External/Plugins/ASCompletion/Completion/ASGenerator.cs]: Does that work when you generate a function for `foo` for `someObj.foo`?

Comment (SlavaRa) [inline External/Plugins/ASCompletion/Completion/ASGenerator.cs]: I will add more tests today

**Pull request closed without merging**